tck/src/main/java/org/eclipse/microprofile/lra/tck/TckLRATypeTests.java

```diff
@@ -0,0 +1,347 @@
+    }
+
+    @Test
+    public void requiredEndWithoutLRATest() {
+        resourceRequest(REQUIRED_WITH_END_PATH, false, 200, MethodLRACheck.NOT_EQUALS, true);
+    }
+
+    @Test
+    public void requiresEndNewWithLRATest() {
+        resourceRequest(REQUIRES_NEW_WITH_END_PATH, true, 200, MethodLRACheck.NOT_EQUALS, true);
+    }
+
+    @Test
+    public void requiresEndNewWithoutLRATest() {
+        resourceRequest(REQUIRES_NEW_WITH_END_PATH, false, 200, MethodLRACheck.NOT_EQUALS, true);
+    }
+
+    @Test
+    public void mandatoryEndWithLRATest() {
+        resourceRequest(MANDATORY_WITH_END_PATH, true, 200, MethodLRACheck.EQUALS, true);
+    }
+
+    @Test
+    public void mandatoryEndWithoutLRATest() {
+        resourceRequest(MANDATORY_WITH_END_PATH, false, 412, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void supportsEndWithLRATest() {
+        resourceRequest(SUPPORTS_WITH_END_PATH, true, 200, MethodLRACheck.EQUALS, true);
+    }
+
+    @Test
+    public void supportsEndWithoutLRATest() {
+        resourceRequest(SUPPORTS_WITH_END_PATH, false, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void notSupportedEndWithRATest() {
+        resourceRequest(NOT_SUPPORTED_WITH_END_PATH, true, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void notSupportedEndWithoutLRATest() {
+        resourceRequest(NOT_SUPPORTED_WITH_END_PATH, false, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void neverWithEndRATest() {
+        resourceRequest(NEVER_WITH_END_PATH, true, 412, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    @Test
+    public void neverWithoutEndLRATest() {
+        resourceRequest(NEVER_WITH_END_PATH, false, 200, MethodLRACheck.NOT_PRESENT, false);
+    }
+
+    /**
+     * Perform a JAX-RS resource request and check the resulting status and whether or not it ran with
+     * the correct LRA context.
+     *
+     * @param path the resource path of the JAX-RS method to invoke
+     * @param startLRA indicates whether or not an active context should be present on the request
+     * @param expectedStatus the expected JAX-RS status code in the response
+     * @param lraCheckType the type of check to perform on the request context and the context specified
+     *                     by the {@link LRA.Type} annotation
+     * @param methodLRAShouldBeActive if true the LRA started by the invoked method should still
+     *                               be active after the resource invocation completes
+     */
+    private void resourceRequest(String path, boolean startLRA, int expectedStatus, MethodLRACheck lraCheckType,
+                                 boolean methodLRAShouldBeActive) {
+        Invocation.Builder target = tckSuiteTarget.path(TCK_LRA_TYPE_RESOURCE_PATH)
+                .path(path).request();
+        URI lra = startLRA ? lraClient.startLRA(null, lraClientId(), lraTimeout(), ChronoUnit.MILLIS) : null;
```

Comment:
.startLRA() doesn't use the client Id and Timeout parameters. Probably needs cleanup.

Reply:
It seems overkill to schedule a timer to cancel the LRA and to cancel the timer when the LRA is closed. Personally I think we should run all LRAs with an indefinite timeout. If a test hangs because of that then the test is brittle and needs rewriting. Do you still want me to manage timers and their cancellation if the LRAs finish early?

Reply:
Well I implemented it anyway (took over 100 lines of code diffs, lol). I didn't squash the diff so that it is clear which bit of code is implementing the timeout behaviour.